### ISSUE TITLE
[Suggestion] Keeping pokemon based on IV rather than CP

### ISSUE THREAD

**Issue by buusc0** [PokemonGoF/PokemonGo-Bot#297]
It's been discovered that pokemons have a hidden IV that determine their max stats, in comparison to CP which reflects current stats (more at: https://www.reddit.com/r/TheSilphRoad/comments/4tupg5/eli5_what_is_iv_and_what_is_perfect/). 

For a lot of players who wish to dedicate more time to the game, IV is likely valued over CP, as CP can be improved with renewable resources (dust and candy) whereas IV is set and can't be changed. This will be particularly important once trading begins, as high IV pokemons will likely sell for high prices, and therefore it'll be great if we can add an option to keep pokemon based on their IV rather than CP.

There's currently two ways to determine the IV of a pokemon without using the API, one by estimating using a formula (https://www.reddit.com/r/TheSilphRoad/comments/4tkk75/updated_iv_calculator_automatically_calculate_ivs/), and another by using a man-in-the-middle proxy to intercept and parse the relevant data (https://www.reddit.com/r/TheSilphRoad/comments/4tqddw/guide_to_determine_exact_ivs_using_mitm_proxy/). Hope this is of some help.

Just a suggestion. Great work guys!

**Comment by boriss579** [comment]
Yest, it will be great if we will have a command that we can config how much perfection percentage we like to keep - like keep all Pokemon with 80% perfection or keep all Pokemon with 13/13/13+ IVs.

**Comment by 05-032** [comment]
The weird thing is, the API will sometimes return some IV values but not others. I've had a Pidgey sometimes have all 3 in the pokemon_data dictionary but one or more might be missing. Does anyone know why this is?

**Comment by 05-032** [comment]
The weird thing is, the API will sometimes return some IV values but not others. Sometimes a pokemon might have all three in the pokemon_data dictionary but other times one or more might be missing. It seems to be random. Does anyone know why this is?

**Comment by buusc0** [comment]
@05-032 I believe it could be because a certain IV is 0. If you take a look at the spreadsheet for the MITM proxy on reddit (https://www.reddit.com/r/TheSilphRoad/comments/4tqddw/guide_to_determine_exact_ivs_using_mitm_proxy/ under Requirements -> 2. A copy of the spreadsheet), you'll see in the output page that there's a snorlax with 15 Att IV, 0 Def IV, and 15 Sta IV.
<img width="806" alt="screen shot 2016-07-22 at 9 07 57 pm" src="https://cloud.githubusercontent.com/assets/20603342/17074851/692b5b1e-5050-11e6-9581-e0071b1b1d2e.png">
This corresponds to an empty entry for the Def IV in the unparsed data (notice how "18:" is missing at row 2400 of the unparsed data).
<img width="327" alt="screen shot 2016-07-22 at 9 10 03 pm" src="https://cloud.githubusercontent.com/assets/20603342/17074862/c090374e-5050-11e6-8a46-ef8038cece4f.png">

**Comment by 05-032** [comment]
Oh, that makes sense. Thanks!

**Comment by 05-032** [comment]
I've added a pull request that should do most of what OP is requesting.
https://github.com/PokemonGoF/PokemonGo-Bot/pull/346

**Comment by buusc0** [comment]
@05-032 Amazing, thanks!

**Comment by buusc0** [comment]
@05-032 Could you possibly add the IV checking ability to the --initial-transfer flag, so that the user can select whether to transfer pokemons based on IV or CP?

**Comment by surfaace** [comment]
IV implemented
